File: app/src/main/java/br/com/onibusapp/onibusapp/ui/mapa/MapsContract.java
package br.com.onibusapp.onibusapp.ui.mapa;

import android.app.Activity;

import com.google.android.gms.maps.model.LatLng;

/**
 * Created by diego on 02/08/2018.
 */

public interface MapsContract {

    interface View {

        void mapaPronto();

        void setLocalizacao(Double lat, Double lnt, Boolean focarLocalizacao);

        Activity getCurrentActivity();

        void limparMapa();

        void addMarker(LatLng latLng, String linha, String prefixo);

    }

    interface Presenter {

        void mapaPronto();

        void setLocalizacao(Double lat, Double lnt, Boolean focarLocalizacao);

        void getLocalizacaoUsuario(Boolean focarLocalizacao);

        void getLocalizacaoOnibus(String linha, Integer sentido, Integer codigoEmpresa);

    }
}

File: app/src/main/java/br/com/onibusapp/onibusapp/ui/mapa/MapsContract.kt
package br.com.onibusapp.onibusapp.ui.mapa

import android.app.Activity

import com.google.android.gms.maps.model.LatLng

/**
 * Created by diego on 02/08/2018.
 */

interface MapsContract {

    interface View {

        fun getCurrentActivity() : Activity

        fun mapaPronto()

        fun setLocalizacao(lat: Double?, lnt: Double?, focarLocalizacao: Boolean?)

        fun limparMapa()

        fun addMarker(latLng: LatLng, linha: String, prefixo: String)

    }

    interface Presenter {

        fun mapaPronto()

        fun setLocalizacao(lat: Double?, lnt: Double?, focarLocalizacao: Boolean?)

        fun getLocalizacaoUsuario(focarLocalizacao: Boolean?)

        fun getLocalizacaoOnibus(linha: String, sentido: Int?, codigoEmpresa: Int?)

    }
}
